de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/FuncElectricalDiagramTest.java

```diff
@@ -0,0 +1,189 @@
+import de.dlr.sc.virsat.model.extension.funcelectrical.model.InterfaceType;
+import de.dlr.sc.virsat.model.extension.funcelectrical.model.InterfaceTypeCollection;
+import de.dlr.sc.virsat.model.extension.ps.model.ConfigurationTree;
+import de.dlr.sc.virsat.model.extension.ps.model.ElementConfiguration;
+import org.junit.Assert;
+
+/**
+ * 
+ * SWTBot test class for testing Funcelectrical Diagrams
+ *
+ */
+public class FuncElectricalDiagramTest extends ASwtBotTestCase {
+	
+	private SWTBotTreeItem repositoryNavigatorItem;
+	private SWTBotTreeItem configurationTree;
+	private SWTBotTreeItem elementConfiguration;
+	private SWTBotGefEditor diagramEditor;
+	
+	private Concept conceptFuncElectrical;
+	
+	private static final int DRAG1_COORDINATES_X = 100;
+	private static final int DRAG1_COORDINATES_Y = 100;
+	
+	private static final int DRAG2_COORDINATES_X = 100;
+	private static final int DRAG2_COORDINATES_Y = 200;
+
+	@Before
+	public void before() throws Exception {
+		super.before();
+		// create the necessary items for the test
+		conceptFuncElectrical = ConceptXmiLoader.loadConceptFromPlugin(de.dlr.sc.virsat.model.extension.funcelectrical.Activator.getPluginId() + "/concept/concept.xmi");
+		repositoryNavigatorItem = bot.tree().expandNode(SWTBOT_TEST_PROJECTNAME, "Repository");
+		configurationTree = addElement(ConfigurationTree.class, conceptPs, repositoryNavigatorItem);
+		elementConfiguration = addElement(ElementConfiguration.class, conceptPs, configurationTree);
+				
+		createNewDiagramForTreeItem(configurationTree, DiagramType.interfaces);
+		diagramEditor = getOpenedDiagramEditorbyTitle("newDiagram");
+	}
+	
+	@Test
+	public void addInterfaceDiagramElementUndoRedoTest() {
+		addElement(Interface.class, conceptFuncElectrical, elementConfiguration);
+		dragTreeItemToDiagramEditor(elementConfiguration, diagramEditor);
+		Assert.assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		undoCommand();
+		Assert.assertFalse(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		redoCommand();
+		Assert.assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+	}	
+
+	@Test
+	public void deleteObjectOutsideDiagramUpdateDiagramTest() {
+		addElement(Interface.class, conceptFuncElectrical, elementConfiguration);
+		dragTreeItemToDiagramEditor(elementConfiguration, diagramEditor);
+		Assert.assertTrue(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+		
+		delete(elementConfiguration);
+		updateActiveDiagram(diagramEditor);
+		Assert.assertFalse(isEditPartPresentInDiagramEditor(diagramEditor, "ElementConfiguration"));
+	}
+	
+	@Test
+	public void dragDropTreeinDiagram() {
+		addElement(Interface.class, conceptFuncElectrical, elementConfiguration);
+		dragTreeItemToDiagramEditor(configurationTree, diagramEditor);
+		Assert.assertEquals(0, diagramEditor.selectedEditParts().size(), 0);
```

Comment:
The assert equals here seems to be the wrong one (it has the float/double signature) but we are checking an int.
In fact, assertTrue(diagramEditor.selectedEditParts().isEmpty()) should be even cleaner.